src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPatternsManager.java

```diff
@@ -1,47 +1,31 @@
 package edu.agh.wfiis.designpatterns.patternsUsageExample;
 
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
 import java.io.File;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
 
+@Service
 public class DesignPatternsManager {
+    private final Set<HavingName> patternNames;
 
-    public DesignPattern create(PatternName patternName) {
-        switch (patternName) {
-            case BUILDER: {
-                DesignPattern pattern = new DesignPattern("1","builder","builder pattern",null, true, false, false);
-                return pattern;
-            }
-            case DECORATOR: {
-                DesignPattern pattern = new DesignPattern("2","decorator","decorator pattern",null, true, false, false);
-                return pattern;
-            }
-            case STRATEGY: {
-                DesignPattern pattern = new DesignPattern("3","strategy","strategy pattern",null, false, false, true);
-                return pattern;
-            }
-            default: {
-                DesignPattern pattern = new UndefinedPattern();
-                return pattern;
-            }
-        }
-    }
-
-    public String provideUsageExample(DesignPattern pattern){
-        String example = pattern instanceof UndefinedPattern? null: pattern.getExample();
-        return example;
+    @Autowired
+    public DesignPatternsManager(Set<HavingName> patternNames) {
+        this.patternNames = patternNames;
     }
 
-    public String refactorCode(DesignPattern pattern, String badCode){
-        String goodCode = pattern instanceof UndefinedPattern? null: pattern.refactor(badCode);
-        return goodCode;
+    public PatternHandler using(DesignPattern designPattern) {
+        return new PatternHandler(designPattern);
     }
 
     public String listPatterns(){
-        return PatternName.values().toString();
-    }
-
-    public boolean areSameType(DesignPattern pattern1, DesignPattern pattern2){
-        return pattern1.isBehavioral() == pattern2.isBehavioral() && pattern1.isCreational() == pattern2.isCreational() && pattern1.isStructural() == pattern2.isStructural();
+        return patternNames.stream()
+                           .map(HavingName::name)
+                           .collect(Collectors.joining("\n"));
     }
 
     public String extractBadCodeFromFile(File file){
```

Comment:
If we' d like to add another extract source kind e.g. extractBadCodeFromEmail , another method should be added, what is inccorect. Moreover, there is only one method to extract from file, if we'd like to change behaviour of extracting file we have to change current implementation, which is in violation of Open/Closed Principle.